rust/src/utils/base64.rs

```diff
@@ -0,0 +1,204 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Shivani Bhardwaj <shivani@oisf.net>
+
+use std::io::{Error, ErrorKind, Result};
+
+fn base64_map(input: u8) -> Result<u8> {
```

Comment:
Option instead of Result ?

Reply:
ok

Reply:
Was doing this and felt that Result makes more sense. An invalid base64 char should mean an error. wdyt?

Reply:
We do not have a rich error containing meaningful data. `Option` thus looks enough... But this is a personal style nit